src/mlpack/methods/ann/ffn.hpp

```diff
@@ -215,6 +217,15 @@ class FFN
   void Forward(arma::mat&& input);
 
   /**
+   * Prepare the network for the given training data.
+   * This function won't actually trigger training process.
+   *
+   * @param predictors Input training variables.
+   * @param responses Outputs results from input training variables.
+   */
+  void SetTrainingData(const arma::mat& predictors, const arma::mat& responses);
```

Comment:
You are right, avoiding code duplication. I think the function name is misleading, it's not necessarily the training set, what you do you think about ``ResetData(...)``?